monai/networks/layers/simplelayers.py

```diff
@@ -27,16 +27,38 @@
 
 class SkipConnection(nn.Module):
     """
-    Concats the forward pass input with the result from the given submodule.
+    Combine the forward pass input with the result from the given submodule::
+
+        --+--submodule--o--
+          |_____________|
+
+    The available modes are ``"cat"``, ``"add"``, ``"mul"``.
     """
 
-    def __init__(self, submodule, cat_dim: int = 1) -> None:
+    def __init__(self, submodule, dim: int = 1, mode: str = "cat") -> None:
+        """
+
+        Args:
+            submodule: the module defines the trainable branch.
+            dim: the dimension over which the tensors are concatenated.
+                Used when mode is ``"cat"``.
+            mode: ``"cat"``, ``"add"``, ``"mul"``. defaults to ``"cat"``.
+        """
         super().__init__()
         self.submodule = submodule
-        self.cat_dim = cat_dim
+        self.dim = dim
+        self.mode = mode
```

Comment:
I think the mode strings should be put into constants in `monai/utils/enums.py`. There can then be a proper check in the constructor that a correct mode was selected, raising ValueError if not. The raise in the `forward` method can then be removed.

Reply:
it seems the enum object doesn't work well with torchscript, @Nic-Ma any idea?

Reply:
It should be working, there's references to it in Pytorch commits, but I'm having problems getting it to work as well.